flink-end-to-end-tests/test-scripts/common_ssl.sh

```diff
@@ -77,7 +77,17 @@ function _set_conf_ssl_helper {
         # -> we need to build it ourselves
         FLINK_SHADED_VERSION=$(cat ${END_TO_END_DIR}/../pom.xml | sed -n 's/.*<flink.shaded.version>\(.*\)<\/flink.shaded.version>/\1/p')
         echo "BUILDING flink-shaded-netty-tcnative-static"
-        git clone https://github.com/apache/flink-shaded.git
+        # Adding retry to git clone, due to FLINK-24971
+        CLONE_COUNTER=1
+        CLONE_RETRIES=4 # Since we start the counter at 1, we are retrying this 3 times
+        CLONE_SLEEP=1
+        until [ $CLONE_COUNTER -ge $CLONE_RETRIES ]
+        do
+            echo "Cloning flink-shaded, attempt number" $CLONE_COUNTER
+            git clone https://github.com/apache/flink-shaded.git && break
+            CLONE_COUNTER=$[$CLONE_COUNTER+1]
+            sleep $CLONE_SLEEP
+        done
```

Comment:
I wonder if we can add that as a general function for reuse on other batch tests...

Reply:
That shouldn't be too hard. I guess the generic function should be in `common.sh` ?